### PR TITLE
update filter + associated collection

### PR DESCRIPTION
**Pull request by nimitsarup**
1. Make filter regex/ignore case
2. Add flag for filter using associated collection

### CODE REVIEW

File: models/interactives.go [hunk 1]
```diff
@@ -40,6 +40,15 @@ type InteractiveUpdate struct {
 	Interactive      Interactive `json:"interactive,omitempty"`
 }
 
+// If getlinkedtocollection is true, i.e visit index page through a collection
+//    disregard other metadata fields and pick only collectionid
+// else
+//    usual filter behaviour
+type InteractiveFilter struct {
+	FilterByAssocCollection bool                 `json:"filter_by_assoc_collection,omitempty"`
```
Comment (mryan321) [inline models/interactives.go]: i dont think we need the prefix - just `AssocCollection`?

Comment (mryan321) [reply models/interactives.go]: not sure on the name too 😄  maybe `LinkingCollection` - the action it will used from?

File: mongo/mongo.go [hunk 2]
```diff
@@ -153,21 +154,32 @@ func (m *Mongo) ListInteractives(ctx context.Context, offset, limit int, modelFi
 // generate filter string depending on data type
 // string eq value
 // array in value(s)
-func generateFilter(model *models.InteractiveMetadata) bson.M {
+func generateFilter(model *models.InteractiveFilter) bson.M {
 	filter := bson.M{}
 	filter["active"] = bson.M{"$eq": true}
-	if model == nil {
+	if model == nil || model.Metadata == nil {
```
Comment (mryan321) [inline mongo/mongo.go:160]: is this right? metadata might be nil if we use the other filter?

File: mongo/mongo.go [hunk 3]
```diff
@@ -153,21 +154,32 @@ func (m *Mongo) ListInteractives(ctx context.Context, offset, limit int, modelFi
...
+		return filter
+	}
+
+	// if filter by collection-id
```
Comment (mryan321) [inline mongo/mongo.go:164]: dunno if we need these comments - the code enough imo 😊  but upto you

Comment (mryan321) [inline mongo/mongo.go:164]: dunno if we need these comments - its quite self documenting 😊

**Review by mryan321 — DISMISSED** [review]
few cosmetic suggestions

File: mongo/mongo.go (hunk 2 above)
Comment (nimitsarup) [reply mongo/mongo.go:160]: metadata needs to be there always ... for both cases..
for linked-collection filter.. the collection-id needs to be in the metadata struct

File: mongo/mongo.go (hunk 3 above)
Comment (nimitsarup) [reply mongo/mongo.go:164]: can i leave them ? :) .. will forget what/why i did in a week

**Review by mryan321 — APPROVED** [review]

**Pull request merged**